spec/index.html

```diff
@@ -8941,13 +8953,120 @@ <h5>Collect <code>FILTER</code> Elements</h5>
 Let FS := empty set
 For each form FILTER(expr) in the group graph pattern
     In expr, replace NOT EXISTS{P} with fn:not(<a href="#defn_evalExists">exists(translate(P)))</a> 
-    In expr, replace EXISTS{P} with <a href="#defn_evalExists">exists(translate(P))</a>
+    In expr, replace EXISTS{P} with prepare(EXISTS{P})
     FS := FS ∪ {expr}
     End
 </pre>
             <p>The set of filter expressions <code>FS</code> is <a href="#sparqlAddFilters">used
                 later</a>.</p>
+
+            <div>
+              <i>Prepare EXISTS and NOT EXISTS</i>
+              <p><b>prepare(EXISTS{P})</b></p>
+
+              <div class="ednote">
+                <p>@@ Scoping rule already done (grammar note).<br/>
+                  Prepare FILTER for exists. 
+                </p>
+                <ul>
+                  <li>A1 = translate(P) without applying simplification</li>
+                  <li>A2 = A1 rewritten to include access current binding</li>
+                  <li>A3 = A2 Remapped hidden-scope variables (if we do that)</li>
+                </ul>
+
+                <div class="defn">
+                  <div>
+                    <b>Definition: Access to the current binding</b>
+                    <p>
+                      During <a href="#sparqlQuery">translation to the SPARQL algebra</a>
+                    </p>
+                  <pre>
+Replace each occurence of `Y` in X where `Y` is one of
+          <a href="#sparqlTranslateBasicGraphPatterns">Basic Graph Pattern</a>,
+          <a href="#sparqlTranslatePathExpressions">Property Path Expression</a>,
+          <a href="#sparqlTranslateGraphPatterns">`Graph(Var, pattern)`</a>,
+          <a href="#https://www.w3.org/TR/sparql12-query/#sparqlTranslateGraphPatterns">Inline Data</a>
+          with `join(Y, BindingInScope())`.</pre>
+                  </div>
+                  <div class="note">
+                    c.f. section <a href="#sparqlTranslateGraphPatterns">Translate Graph Patterns</a>
+                    where an empty basic graph pattern start any 
+                    <a href="#rGroupGraphPattern">GroupGraphPattern</a>.
+                    It happens before the <a href="#sparqlSimplification">simplification step</a>.
+                  </div>
+                </div>
+            
+                <div class="ednote" id="note-BindingInScope">
+                  <p>
+                    One way to provide `BindingInScope` is to change `eval` to also have the current row
+                    as an argument or a "null" token. . Normally this the "null" token. This would be set in
+                    eval-filter.
+                  </p>
+                  <p>
+                    Another way is to have a global (to the execution) variable. In eval-exists, the
+                    old value is recorded, the global set to the new value, and reset on exit - this forms a
+                    stack for nested EXISTS.
```

Comment:
It's not clear what is `reset on exit`. I think, maybe, replace that phrase with `the global reset to the old value on exit`, as suggested below. The hyphen should become either a full stop (as suggested below) or an em-dash (which I think doesn't work as well, here).
```suggestion
                    Another way is to have a global (to the execution) variable. In eval-exists, the
                    old value is recorded, the global set to the new value, and the global reset to the old value on exit. This forms a
                    stack for nested `EXISTS`.
```